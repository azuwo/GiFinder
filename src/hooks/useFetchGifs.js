import { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";

const useFetchGifs = (category, gifsPerPage) => {
  const [state, setState] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    getGifs(category, gifsPerPage).then((imgs) => {
      setState({
        data: imgs,
        loading: false,
      });
    });
  }, [category, gifsPerPage]);

  return state;
};

export { useFetchGifs };
