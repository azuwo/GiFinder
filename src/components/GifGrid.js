import React from 'react'
import PropTypes from 'prop-types'
import { useFetchGifs } from '../hooks/useFetchGifs'
import GifGridItem from './GifGridItem'

const GifGrid = ({category}) => {

    const {data:images} = useFetchGifs(category)

    return (
        <>
            <h3>{category}</h3>
           { <div className = 'card-grid'>
                {
                    images.map( img  =>(
                        <GifGridItem
                            key= {img.id}
                            { ...img }
                        />
                    ))
                }
            </div>}
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid
