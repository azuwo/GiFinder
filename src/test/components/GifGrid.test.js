import { shallow } from 'enzyme';
import React from 'react'
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')

describe('Test in GifGrid', () => {
    const category = 'Goku'

    test('should show component', () => {

        useFetchGifs.mockReturnValue({
            data:[],
            loading: false
        })
        const wrapper = shallow(<GifGrid category={category}/>)
        expect(wrapper).toMatchSnapshot()
    });

    test('should show items when images from useFechGifs are loaded', () => {
        const gifs = [{
            id:'ABC',
            url:'https://localhost/cualquiercosa/cosa.jpg',
            title:'Cualquier cosa'
        }]

        useFetchGifs.mockReturnValue({
            data:gifs,
            loading: false
        })
        const wrapper = shallow(<GifGrid category={category}/>)
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
});