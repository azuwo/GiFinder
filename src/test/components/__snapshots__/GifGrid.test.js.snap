// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifGrid should show component 1`] = `
<Fragment>
  <h3>
    Goku
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;
